pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,500 @@
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+        """
+        for edge in ebunch:
+            self.add_edge(edge[0], edge[1])
+
+    def intra_slice(self, timeslice):
+        """
+        returns the intra slice edges present in a single time slice.
+
+        Parameters:
+        ----------
+        timeslice: integer value. should be ranging from 0
+                   to the number of the slices present in
+                   the bayesian network.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+        >>> dbn.intra_slice(0)
+        [edge_type(start=('D', 0), end=('G', 0)),
+        edge_type(start=('I', 0), end=('G', 0)),
+        edge_type(start=('G', 0), end=('L', 0))]
+        """
+        if not isinstance(timeslice, int):
+            raise ValueError('the timeslice is not an integer value')
+        if timeslice < 0:
+            raise ValueError('negative values for a timeslice are not permissible')
+        if timeslice > self.timestate:
+            raise ValueError('the bayesian network does not contain the intra slice edges of this timeslice')
+        return list(filter(lambda x: x.start[1] == x.end[1] == timeslice, self.intra_edges))
+
+    def inter_slice(self, timeslice):
+        """
+        returns the inter-slice edges in which the edges end in
+        the given time slice
+        
+        Parameters:
+        ----------
+        timeslice: integer value. should be ranging from 0
+                   to the number of the slices present in
+                   the bayesian network.
+
+        EXAMPLE:
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1)))])
+        >>> dbn.inter_slice(1)
+        [edge_type(start=('D', 0), end=('D', 1)),
+        edge_type(start=('I', 0), end=('I', 1))]
+        """
+        if not isinstance(timeslice, int):
+            raise ValueError('the timeslice is not an integer value')
+        if timeslice < 0:
+            raise ValueError('negative values for a timeslice are not permissible')
+        if timeslice > self.timestate:
+            raise ValueError('the bayesian network does not contain the inter slice edges of this timeslice')
+        return list(filter(lambda x: x.end[1] == timeslice, self.inter_edges))
+
+    def add_dbn_cpds(self, timeslice, *cpds):
```

Comment:
Please ensure consistency in naming methods, it should be ```add_cpds``` for directed graphical models and ```add_factors``` for undirected graphical models.